test/regression/sql/time_bucket.sql

```diff
@@ -0,0 +1,52 @@
+-- All functions should automatically use duckdb
+set duckdb.execution = false;
+
+SELECT time_bucket('1 day'::interval, '2023-01-15'::date);
+
+SELECT time_bucket('1 month'::interval, '2023-01-15'::date, '2022-12-25'::date);
+
+SELECT time_bucket('1 week'::interval, '2023-01-15'::date, '2 days'::interval);
+
+SELECT time_bucket('1 hour'::interval, '2023-01-15 14:30:00'::timestamp);
+
+SELECT time_bucket('4 hours'::interval, '2023-01-15 14:30:00'::timestamp, '1 hour'::interval);
+
+SELECT time_bucket('1 day'::interval, '2023-01-15 14:30:00'::timestamp, '2022-12-25 06:00:00'::timestamp);
+
+SELECT time_bucket('12 hours'::interval, '2023-01-15 14:30:00+00'::timestamp with time zone);
+
+SELECT time_bucket('1 day'::interval, '2023-01-15 14:30:00+00'::timestamp with time zone, '6 hours'::interval);
+
+SELECT time_bucket('1 week'::interval, '2023-01-15 14:30:00+00'::timestamp with time zone, '2022-12-25 12:00:00+00'::timestamp with time zone);
+
+SELECT time_bucket('1 day'::interval, '2023-01-15 14:30:00+00'::timestamp with time zone, 'America/New_York');
+
+SELECT time_bucket('1 day'::interval, r['date_col'])
+FROM duckdb.query('SELECT DATE ''2023-01-15'' AS date_col') r;
```

Comment:
Instead of escaping the quotes by duplicating them let's use the double dollar syntax to create these SQL strings, so escaping is not necessary.
```suggestion
FROM duckdb.query($$ SELECT DATE '2023-01-15' AS date_col $$) r;
```